cs-multi-shell/src/app/home/home.page.ts

```diff
@@ -1,17 +1,33 @@
-import { Component } from '@angular/core';
-import { IonHeader, IonToolbar, IonTitle, IonContent, IonButton } from '@ionic/angular/standalone';
+import { Component, inject } from '@angular/core';
+import { IonHeader, IonToolbar, IonTitle, IonContent, IonButton, IonButtons } from '@ionic/angular/standalone';
+import { AuthenticationService } from '../authentication.service';
 
 @Component({
   selector: 'app-home',
   templateUrl: 'home.page.html',
   styleUrls: ['home.page.scss'],
   standalone: true,
-  imports: [IonButton, IonHeader, IonToolbar, IonTitle, IonContent],
+  imports: [IonButtons, IonButton, IonHeader, IonToolbar, IonTitle, IonContent],
 })
 export class HomePage {
+  private authenticationService = inject(AuthenticationService);
   constructor() { }
 
-  open(app: string) {
+  async open(app: string) {
+    const token = await this.authenticationService.decodeToken();
```

Comment:
Note: We can decode the token to get things like the unique userid, email address etc of the user. You could also pass token info to the app you are launching or leave it up to the app to read the auth information from the vault.